### ISSUE TITLE
Lab 4.1: Explore Classification - instructs us to use an env group for classification

### ISSUE THREAD

**Issue by zoojar** [puppetlabs/ilt-practitioner-issues#54]
## Describe the Bug
Lab 4.1: Explore Classification - instructs us to use an Environment node group for classification - against [best practices for classifying node groups](https://puppet.com/docs/pe/2019.8/grouping_and_classifying_nodes.html#creating_node_groups);

> - Environment node groups assign environments to nodes, such as test, development, or production.
> - Classification node groups assign classification data to nodes, including classes, parameters, and variables.

1. Classify and enforce the userprefs module
2. Return to the PE Console (linked on the welcome page).
3. In your studentN-env environment, click on the Configuration tab.
4. Click the Refresh link to reload the class definitions in the PE console.
5. Add the userprefs class by typing its name into the Add new class text box.
6. Add parameters to the class using the dropdown list.
-- Choose an editor by typing in the parameter value box.
7. Choose from mg (a tiny Emacs clone), nano, or vim .
-- Choose a login shell by typing into the parameter text box.
-- Choose from bash or zsh .
9. Commit your changes by pressing the button.
9. Enforce your configuration on your Linux agent.
-- puppet agent -t

### Now create a Classification Group:
1. In the left menu, click Classification.
2. Click Add group at the top.
3. Enter the following (where N is your student number):
-- Parent name: studentN-env
-- Group name: studentN-hosts
-- Environment: studentN
-- Environment Group: un-checked
-- Description: This group assigns classification for studentN
-- Click Add.
4. Now that the group has been created, expand the All Environments group by clicking the plus sign (+) to the left of the name, and then click your new studentN-hosts group.
5. On the Rules tab, navigate to the section that says Pin specific nodes to the group.
6. Click Node name and select only your Linux machine from the list.
7. Click Pin node.
8. Click Commit X changes in the lower-right corner.

### Classify and enforce the userprefs module

1. Return to the PE Console (linked on the welcome page).
2. In your studentN-hosts node group, click on the Configuration tab.
3. Click the Refresh link to reload the class definitions in the PE console.
4. Add the userprefs class by typing its name into the Add new class text box.
5. Add parameters to the class using the dropdown list.
-- Choose an editor by typing in the parameter value box.
6. Choose from mg (a tiny Emacs clone), nano, or vim .
-- Choose a login shell by typing into the parameter text box.
-- Choose from bash or zsh .
7. Commit your changes by pressing the button.
8. Enforce your configuration on your Linux agent.
-- puppet agent -t

---

## Steps to Reproduce
n/a

## Environment
n/a

## Additional Context
n/a

**Comment by josephoaks** [comment]
@zoojar This lab is done to show classification groups, later lab expands on this to change and create environment groups and classification groups. The purpose of this is many students do this in their environments already and we show them how to change and follow best practice.

**Comment by zoojar** [comment]
Hey! With that in mind, we should at least recognize that is an effective way of confusing people and causing doubt - especially those who are unfamiliar or struggle with the concepts.

**Comment by josephoaks** [comment]
I would be all for showing them the best practice sooner than we do in Lab 12, however we get a ton of students that come from environments that do it this way already. In my classes I tell them such and that we will show the a better way to do it later, and right now for the sake of the that are already using this method we get to show them what not to do. This was modified about a year ago to updated the Lab 12 section which changed.

Maybe we should put a callout or a hint in there saying something like this is what a lot of environments start out like and you maybe familiar or use this method but we will show a better way, best practices later?

**Comment by zoojar** [comment]
> In my classes I tell them such and that we will show the a better way to do it later, and right now for the sake of the that are already using this method we get to show them what not to do.

If the reason is to demonstrate what not to do then would it make more sense to demonstrate basic classification first (using a classification node group) to perhaps apply classes in production, and then later demonstrate how we solve the now problem of needing to be able to test our changes in our branch by taking it further and creating our environment node group?

The overall feedback received when asking about classification (in both Practitioner and GSWP) is that 9 out of 10 students have no idea a) how many types of node group there are, and b) what they are both used for. Showing them a class being added to an environment group then creates a whole sense of frustration and confusion - since just being told the exact opposite!


> Maybe we should put a callout or a hint in there saying something like this is what a lot of environments start out like and you maybe familiar or use this method but we will show a better way, best practices later?

Adding a hint (more information!) means one more thing we add to the student's cognitive load which is already well over capacity by this point (taking into account all the other many concepts they are still juggling).

Alternatively it might benefit to simplify it even further while also demonstrating the use of agent-specified environments which is the standard anyway:

1. Use only 2 environment groups - Production & Agent-Specified.
2. site.pp to handle the classification.
3. Have the class run puppet in their specified environment (via console perhaps) when they need to test their changes (standard canary style).

It's helpful to understand the point of using the console to visualize the work we are doing but being Practitioner it would make more sense to align with best practices - as we know, controlling and changing things through the console [classification-via-the-pe-classifier](https://github.com/puppetlabs/best-practices/blob/master/classification-and-data-pattern.md#classification-via-the-pe-classifier) means we don't get the full benefit of version control and ci/cd methodologies - opening/closing the path to cd4pe.

**Comment by jessereynolds** [comment]
I agree with @zoojar here. We should not be demonstrating things that are against best practice in order to then show the folly of our ways later on. Start as you mean to go on. 

Environment groups and classifications groups should be kept separated. When first doing this, it can be explained to any confused students at that time, rather than confusing the whole class.
